strapi/strapi
# Repeatable Component: expanded option

### What does it do?

Adds an option for repeatable components to be always expanded without requiring to do extra clicks to show contents. It helps to keep all data in front of the eyes when you need to analyze some data with lots of tiny single-line records. Ideally it would be a table, but I don't have enough knowledge of Strapi architecture to implement table just yet.

Describe the technical changes you did:
- added option to strapi-plugin-content-type-builder/advancedForm
- renders same as if collapsed
- all components are not expanded by default so it does not break any default logic

![image](https://user-images.githubusercontent.com/3128402/117228073-f787f100-ae20-11eb-90a6-529b9589c04f.png)

When `Expanded` is on

![image](https://user-images.githubusercontent.com/3128402/117225537-3dda5180-ae1b-11eb-9f9f-15e59f56a846.png)

### Why is it needed?

We're working on a game which contains lots of tiny configurations that should be in a plain sight. Consider following example,
where each NPC has some loot to drop.

**Default View**

![image](https://user-images.githubusercontent.com/3128402/117226780-12a53180-ae1e-11eb-871a-31bce80d8128.png)

**Expanded View**

![image](https://user-images.githubusercontent.com/3128402/117226229-dae9ba00-ae1c-11eb-9ad5-61a12c08b7c5.png)

### How to test it?

- `yarn develop` in `examples/getstarted`
- `yarn develop` in `packages/strapi-admin`
- go to [Content Type Builder > Menusection > Dishes > Edit > Advanced Settings > Check 'Expanded'](http://localhost:4000/admin/plugins/content-type-builder/content-types/application::menusection.menusection?actionType=edit&attributeName=dishes&attributeType=component&kind=collectionType&forTarget=contentType&modalType=attribute&settingType=advanced&step=2&targetUid=application::menusection.menusection&header_label_1=menusection&header_icon_name_1=component&header_icon_isCustom_1=false&header_label_2=dishes&header_icon_isCustom_2=false&header_icon_name_3=component&header_icon_isCustom_3=false&header_icon_isCustom_4=false&header_icon_isCustom_5=false)

### Related issue(s)/PR(s)

No related issues or PR(s) found.
